### ISSUE TITLE
Filter devices by structureId?

### ISSUE THREAD

**Issue by laverdet** [potmat/homebridge-google-nest-sdm#174]
**Have you read the [FAQ](https://github.com/potmat/homebridge-google-nest-sdm#faq)?**

Yes

**Have you searched [existing issues](https://github.com/potmat/homebridge-google-nest-sdm/issues?q=is%3Aissue)?**

Yes (searched for "structure")

**Have you searched [existing discussions](https://github.com/potmat/homebridge-google-nest-sdm/discussions?discussions_q=)?**

Yes (searched for "structure")

**Describe the bug**

There doesn't seem to be a way to filter accessories by "structure" or building / location. So if you have more than one location configured on the same Nest account, then all of those accessories will be visible to HomeKit. Homebridge doesn't seem to have a way to filter extraneous devices, this must be built per-plugin. homebridge-nest has this feature via the [`structureId` flag](https://github.com/chrisjshull/homebridge-nest?tab=readme-ov-file#configuration).

In the Nest OAuth workflow you can authorize the app so that it can only see one structure, but if you want to set up Homebridge in more than one location then you have a problem. When you go through the OAuth workflow again on the same account, the devices under the second structure will show up on both Homebridge instances.

Note: I tried to set up a second Device Access project for the second location as a workaround, but Google won't let you enable pub/sub right now. It's just broken on their end. Some users are complaining about this here https://www.reddit.com/r/homeassistant/comments/1dmkpu6/failure_to_create_project_device_access_console/

**Attach Logs**
n/a

**Screenshots**
n/a

**Environment**
n/a

**Additional context**
Thank you for your work on this plugin!

**Comment by potmat** [comment]
Just so I'm clear, you mean that structureId is a configuration parameter and if set it will filter all devices keeping only those with a matching structure id?

**Comment by laverdet** [comment]
Yes, that is the behavior of homebridge-nest. In your Homebridge configuration you specify the Nest `structureId` which corresponds to the physical location of this Homebridge instance. The plugin will then only advertise the devices which belong to that location.

**Comment by laverdet** [comment]
I'm running these modifications locally. It works for my case. I had to add console logs to find the given structure id which is a base64 string ~107 characters in length.

```diff
diff --git a/src/Config.ts b/src/Config.ts
index 52d19a5..e2471c8 100644
--- a/src/Config.ts
+++ b/src/Config.ts
@@ -7,5 +7,6 @@ export type Config = {
     gcpProjectId?: string,
     vEncoder?: string,
     showFan?: boolean,
-    fanDuration?: number
+    fanDuration?: number,
+    structureId?: string,
 }
diff --git a/src/sdm/Api.ts b/src/sdm/Api.ts
index e953d83..c525e72 100644
--- a/src/sdm/Api.ts
+++ b/src/sdm/Api.ts
@@ -17,6 +17,7 @@ export class SmartDeviceManagement {
     private pubSubClient: pubsub.PubSub | undefined;
     private subscription: pubsub.Subscription | undefined;
     private projectId: string;
+    private structureId: string | undefined;
     private log: Logger;
     private devices: Device[] | undefined;
     private subscribed = true;
@@ -29,6 +30,7 @@ export class SmartDeviceManagement {
             config.clientSecret
         );
         this.projectId = config.projectId;
+        this.structureId = config.structureId;
         this.oauth2Client.setCredentials({
             refresh_token: config.refreshToken
         });
@@ -98,6 +100,11 @@ export class SmartDeviceManagement {
             this.log.debug('Receieved list of devices: ', response.data.devices)
 
             this.devices = _(response.data.devices)
+                .filter(
+                    this.structureId === undefined ?
+                        () => true :
+                        device => (device.parentRelations?.some(relation => relation.parent?.includes(`structures/${this.structureId}`)) ?? false)
+                )
                 .filter(device => device.name !== null)
                 .map(device => {
                     switch (device.type) {
```

 Note that this doesn't work, which is a bit surprising. It throws a 400 or 404:
```js
const response = await this.smartdevicemanagement.enterprises.devices.list({parent: `enterprises/${this.projectId}/structures/${this.structureId}`})
```

I didn't see an easy way to list only devices belonging to a given structure. Not sure how Google's API is supposed to scale to insane power users.
